cmd/clusterctl/client/repository/repository_github.go

```diff
@@ -216,9 +220,24 @@ func (g *gitHubRepository) getVersions() ([]string, error) {
 
 	// get all the releases
 	// NB. currently Github API does not support result ordering, so it not possible to limit results
-	releases, _, err := client.Repositories.ListReleases(context.TODO(), g.owner, g.repository, nil)
-	if err != nil {
-		return nil, g.handleGithubErr(err, "failed to get the list of releases")
+	var releases []*github.RepositoryRelease
+	var retryError error
+	_ = wait.PollImmediate(retryableOperationInterval, retryableOperationTimeout, func() (bool, error) {
+		var listReleasesErr error
+		releases, _, listReleasesErr = client.Repositories.ListReleases(context.TODO(), g.owner, g.repository, nil)
+		if listReleasesErr != nil {
+			// return immediately if we are rate limited
+			if _, ok := listReleasesErr.(*github.RateLimitError); ok {
+				return false, listReleasesErr
```

Comment:
```suggestion
				return false, g.handleGithubErr(listReleasesErr, "failed to get the list of releases")
```

This makes sure that we show the properly formatted error if we hit a rate limiting error.

Same comment at the other places.

Reply:
done, thx!

I'll look into lint errors tomorrow a.m. PST, not sure what's happening here...


```
$ make lint
/Users/jackfrancis/go/src/sigs.k8s.io/cluster-api/hack/tools/bin/golangci-lint run -v 
panic: load embedded ruleguard rules: rules/rules.go:13: can't load fmt

goroutine 1 [running]:
github.com/go-critic/go-critic/checkers.init.22()
	github.com/go-critic/go-critic@v0.6.2/checkers/embedded_rules.go:46 +0x4b4
make: *** [lint] Error 2
```